### PR TITLE
Release Modifications

### PR DESCRIPTION
**Pull request by atharva-2001**
### :pencil: Description

**Type:** | :roller_coaster: `infrastructure`

Tasks being addressed in this PR-
1. Changelog in release summary
2. Updates release action version
3. Adds arm lockfile to release assets

Sample- https://github.com/atharva-2001/tardis/releases/tag/release-2024.09.02
The items here are large since I did not have a release on my fork since Aug 1.

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Review by andrewfullard — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
           python-version: "3.x"
 
       - name: Install setuptools_scm
-        run: pip install 'setuptools_scm<7'
+        run: pip install "setuptools_scm<7" git-cliff
```
Comment (wkerzendorf) [inline .github/workflows/release.yml:31]: why does setuptools_scm and git-cliff not live in the lock file?

Comment (atharva-2001) [reply .github/workflows/release.yml:31]: We do, but I'm afraid we don't have git cliff in the lock file. So we have to either install the complete environment and on top of that install git cliff, or just install these two.

**Comment by andrewfullard** [comment]
Update PR once git-cliff is in lockfile

**Comment by atharva-2001** [comment]
TODO: put zenodo link in the changelog

**Review by KasukabeDefenceForce — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**